src/prefect/events/clients.py

```diff
@@ -0,0 +1,57 @@
+import abc
+from types import TracebackType
+from typing import List, Optional, Type
+
+from .schemas import Event
+
+
+class EventsClient(abc.ABC):
+    """The abstract interface for a Prefect Events client"""
+
+    @abc.abstractmethod
+    async def emit(self, event: Event) -> None:
+        ...
+
+    async def __aenter__(self) -> "EventsClient":
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[Exception]],
+        exc_val: Optional[Exception],
+        exc_tb: Optional[TracebackType],
+    ) -> "EventsClient":
+        return self
```

Comment:
The return value has a meaning here, I don't think you want to return self.

Reply:
Oops, good call!

Reply:
Oof, that's _really_ bad, it would be a truthy value, thus quietly swallowing exceptions ha!  Thank you!